egs/thchs30/s5/local/dae/add-noise-mod.py

```diff
@@ -0,0 +1,153 @@
+#!/usr/bin/env python
+
+from __future__ import print_function
+import optparse
+import random
+import bisect
+import re
+import logging
+import wave
+import math
+import struct
+import sys
+import os
+
+try:
+  import pyximport; pyximport.install()
+  from util import *
+except:
+  print("Cython possibly not installed, using standard python code. The process might be slow", file=sys.stderr)
+
+  def energy(mat):
+    return float(sum([x * x for x in mat])) / len(mat)
+
+  def mix(mat, noise, pos, scale):
+    ret = []
+    l = len(noise)
+    for i in xrange(len(mat)):
+        x = mat[i]
+        d = int(x + scale * noise[pos])
+        #if d > 32767 or d < -32768:
+        #    logging.debug('overflow occurred!')
+        d = max(min(d, 32767), -32768)
+        ret.append(d)
+        pos += 1
+        if pos == l:
+            pos = 0
+    return (pos, ret)
+
+
+def dirichlet(params):
+    samples = [random.gammavariate(x, 1) if x > 0 else 0. for x in params]
+    samples = [x / sum(samples) for x in samples]
+    for x in xrange(1, len(samples)):
+        samples[x] += samples[x - 1]
+    return bisect.bisect_left(samples, random.random())
+
+def wave_mat(wav_filename):
+    f = wave.open(wav_filename, 'r')
+    n = f.getnframes()
+    ret = f.readframes(n)
+    f.close()
+    return list(struct.unpack('%dh' % n, ret))
+
+def num_samples(mat):
+    return len(mat)
+
+def scp(scp_filename):
+    with open(scp_filename) as f:
+        for l in f:
+            yield tuple(l.strip().split())
+
+def wave_header(sample_array, sample_rate):
```

Comment:
standard python library contains a module called "wave" which could be used instead of wav handling code here. Would be possibly more portable but I don't particularly care if you change it or not.